packages/runtime/src/state/react-builder-preview.ts

```diff
@@ -519,6 +520,35 @@ function measureBoxModelsMiddleware(): Middleware<Dispatch, State, Dispatch> {
     }
 }
 
+function registerDocumentsMiddleware(): Middleware<Dispatch, State, Dispatch> {
+  return ({ dispatch }: MiddlewareAPI<Dispatch, State>) =>
+    (next: ReduxDispatch<Action>) => {
+      return (action: Action): Action => {
+        switch (action.type) {
+          case ActionTypes.REGISTER_DOCUMENT: {
+            if (action.payload.builderInitiated) break
+            const { document } = action.payload
+            dispatch(
+              registerBuilderDocument({
+                ...document,
+                locale: document.locale ?? null,
+              }),
+            )
+            break
+          }
```

Comment:
Looking at this here seems a bit weird to me. We have a middleware that is "rewriting" actions. But if the action is being dispatched by the host, couldn't we `dispatch(registerBuilderDocument(...))` directly instead and avoid the rewrite? And if it's being dispatched from the builder, well `dispatch(registerDocument(...))` should do it. And now since the host and the builder use a different action wouldn't there be no ping-pong and also there would be no need for this "rewrite" middleware?